core/src/mender-client.c

```diff
@@ -1049,6 +1086,20 @@ mender_prepare_new_provides(mender_artifact_ctx_t *mender_artifact_ctx, char **n
         }
     }
 
+    /* Get artifact name from provides */
+    for (mender_key_value_list_t *item = mender_artifact_ctx->artifact_info.provides; NULL != item; item = item->next) {
+        if (StringEqual("artifact_name", item->key)) {
+            *artifact_name = item->value;
+            break;
+        }
+    }
+
+    if (NULL == *artifact_name) {
+        mender_log_error("No artifact name found in provides");
+        mender_utils_free_linked_list(stored_provides);
```

Comment:
`stored_provides` seems to be leaked by this function unless this `if` is executed.

Reply:
`stored_provides` is freed after being combined with `new_provides` in `mender_filter_provides()`

Reply:
Worth a comment.